core/src/avm2/verify.rs

```diff
@@ -131,16 +133,23 @@ pub fn verify_method<'gc>(
                 Err(_) => unreachable!(),
             };
 
-            for exception in body.exceptions.iter() {
+            for (exception_index, exception) in body.exceptions.iter().enumerate() {
                 // If this op is in the to..from and it can throw an error,
                 // add the exception's target to the worklist.
                 if exception.from_offset as i32 <= previous_position
                     && previous_position < exception.to_offset as i32
                     && op_can_throw_error(&op)
-                    && !seen_targets.contains(&(exception.target_offset as i32))
                 {
-                    worklist.push(exception.target_offset);
-                    seen_targets.insert(exception.target_offset as i32);
+                    if !seen_targets.contains(&(exception.target_offset as i32)) {
+                        worklist.push(exception.target_offset);
+                        seen_targets.insert(exception.target_offset as i32);
+                    }
+
+                    // Keep track of all the valid exceptions, and only verify
+                    // them- this is more lenient than avmplus, but still safe.
+                    if !seen_exception_indices.contains(&exception_index) {
```

Comment:
I think the entry() API would let you avoid a double lookup (contains -> insert)

Reply:
This is a `HashSet` though